File: plugins/aref/fixtures/go/cyclic.go
package fixture

// CyclicAlpha calls CyclicBeta which calls CyclicAlpha.
// Single-package mutual recursion substitutes for inter-package cycle
// in this minimal fixture. Multi-package fixture is out of scope for v0.1.
func CyclicAlpha(n int) int {
	if n <= 0 {
		return 0
	}
	return CyclicBeta(n-1) + 1
}

func CyclicBeta(n int) int {
	if n <= 0 {
		return 0
	}
	return CyclicAlpha(n-1) + 2
}

File: plugins/aref/fixtures/rust/src/cyclic.rs
//! Rust prevents module cycles at compile time but workspace crate cycles are possible.
//! For single-crate fixture we simulate a cycle through mutual recursion between modules
//! via a shared module boundary.

pub mod alpha {
    use super::beta;

    pub fn a_fn(n: i32) -> i32 {
        if n <= 0 { return 0; }
        beta::b_fn(n - 1) + 1
    }
}

pub mod beta {
    use super::alpha;

    pub fn b_fn(n: i32) -> i32 {
        if n <= 0 { return 0; }
        alpha::a_fn(n - 1) + 2
    }
}
